File: frontend/src/constants/websocket.ts
/**
 * WebSocket-related constants for the frontend
 */

// Message types for frontend-backend communication
export const WS_MESSAGE_TYPES = {
	STATUS_UPDATE: 'status_update',
	COUNCIL_INITIALIZED: 'council_initialized',
	DOCUMENT_AUDIT_STARTED: 'document_audit_started', 
	DOCUMENT_AUDIT_COMPLETED: 'document_audit_completed',
	AUDIT_STARTED: 'audit_started',
	AUDIT_COMPLETED: 'audit_completed',
	ERROR: 'error',
	ERROR_OCCURRED: 'error_occurred',
	SYSTEM_ALERT: 'system_alert',
	HEARTBEAT: 'heartbeat',
	HEARTBEAT_ACK: 'heartbeat_ack',
	PING: 'ping',
	PONG: 'pong',
} as const

// WebSocket connection configuration
export const WS_CONFIG_DEFAULTS = {
	RECONNECT_INTERVAL: 3000,
	MAX_RECONNECT_ATTEMPTS: 10,
	HEARTBEAT_INTERVAL: 30000,
	MESSAGE_QUEUE_SIZE: 100,
	MAX_RECONNECT_DELAY: 30000,
	INITIAL_RECONNECT_DELAY: 100,
} as const

// Message priorities
export const MESSAGE_PRIORITIES = {
	LOW: 1,
	MEDIUM: 2, 
	HIGH: 3,
} as const

// Activity status mappings
export const ACTIVITY_STATUS_MAP = {
	reviewing: 'reviewing',
	responding: 'responding', 
	debating: 'debating',
	questioning: 'questioning',
	idle: 'idle',
} as const

// Provider mappings
export const PROVIDER_MAP = {
	openai: 'openai',
	anthropic: 'anthropic',
	google: 'google', 
	openrouter: 'openrouter',
} as const

File: src/llm_council/constants/websocket.py
"""WebSocket-related constants for the LLM Council platform."""

# WebSocket message types
WS_MESSAGE_TYPES = {
    "STATUS_UPDATE": "status_update",
    "COUNCIL_INITIALIZED": "council_initialized", 
    "DOCUMENT_AUDIT_STARTED": "document_audit_started",
    "DOCUMENT_AUDIT_COMPLETED": "document_audit_completed",
    "AUDIT_COMPLETED": "audit_completed",
    "ERROR": "error",
    "HEARTBEAT": "heartbeat",
    "PING": "ping",
    "PONG": "pong",
    "HEARTBEAT_ACK": "heartbeat_ack",
}

# WebSocket connection settings
WS_CLOSE_CODES = {
    "NORMAL": 1000,
    "MANUAL_RECONNECT": 1000,
    "SERVICE_DESTROYED": 1000,
}

# Default timeouts and intervals (in seconds)
WS_TIMEOUTS = {
    "HEARTBEAT_INTERVAL": 30.0,
    "RECONNECT_BASE_INTERVAL": 3.0,
    "MAX_RECONNECT_INTERVAL": 30.0,
    "MAX_RECONNECT_ATTEMPTS": 10,
    "MESSAGE_QUEUE_SIZE": 100,
}
